### PR TITLE
Set layout node names from display profiles

### LINKED ISSUE #6475: Issue when adding GUI layout.
### Expected behaviour
Layout added without any error.

### Actual behaviour
Layout added with error and I can't do anything with it like in the attached video.

### Steps to reproduce
Add a layout.

https://user-images.githubusercontent.com/11591738/158977049-0a51777b-a018-4275-a4bb-5bd2bb200daf.mov

<hr/>

<table>
<tr><td>Build time</td><td>2022-03-15T14:41:06.884849</td></tr>
<tr><td>Defold channel</td><td>editor-alpha</td></tr>
<tr><td>Defold editor sha</td><td>cb4cc114221bf2bdabb1dd3c988fb333dc1e836a</td></tr>
<tr><td>Defold engine sha</td><td>0e77ba11ac957ee01878bbde2e6ac0c9fae6dc01</td></tr>
<tr><td>Defold version</td><td>1.3.0<

/td></tr>
<tr><td>Error</td><td><a href='https://sentry.io/organizations/defold/issues/?query=id%3A"63ed1f5a7c984d9ab29152229bb612d7"'>63ed1f5a7c984d9ab29152229bb612d7</a></td></tr>
<tr><td>GPU</td><td>Intel Iris Pro OpenGL Engine</td></tr>
<tr><td>GPU Driver</td><td>2.1 INTEL-12.10.12</td></tr>
<tr><td>Java version</td><td>11.0.1+13</td></tr>
<tr><td>OS arch</td><td>x86_64</td></tr>
<tr><td>OS name</td><td>Mac OS X</td></tr>
<tr><td>OS version</td><td>10.14.6</td></tr>
<table>

### PR DESCRIPTION
**Pull request by vlaaad**
This fixes a crash which happens when adding a layout to a gui scene.

Fixes #6475 

Note: the previous implementation of the `add-handler-info` label
returned a single tuple with information about a single context menu
option. Gui's LayoutsNode has many add options — one for every not yet
defined display profile. I cannot change it to always return
a collection of tuples, because this is label is also used in spine
extension, and since we can't force our users to update, such a change
will break existing users. So I decided to allow both a single tuple and
a collection of tuples to preserve compatibility.

Note: there is a `handler-infos` input label that collects all handler
info tuples from GuiSceneNode's children. Due to a data type change of
`add-handler-info` I also added a `handler-infos` output that ensures
that it's a flat vector of info tuples. My concern is: `:handler-infos`
are completely unused. Should we remove them altogether?

### CODE REVIEW

File: editor/src/clj/editor/gui.clj [hunk 1]
```diff
@@ -2106,16 +2106,27 @@
                   (when select-fn
                     (select-fn [node]))))))
 
+(defn- unused-display-profiles [gui-scene-node-id]
+  (g/with-auto-evaluation-context ec
+    (let [layouts (->> (g/node-value gui-scene-node-id :layout-msgs ec)
+                       (map :name)
+                       set)]
```
Comment (matgis) [inline editor/src/clj/editor/gui.clj:2113]: Transduce `(into #{})`?

File: editor/src/clj/editor/gui.clj [hunk 2]
```diff
@@ -2349,6 +2365,8 @@
   (input textures-node g/NodeID) ; for tests
   (input particlefx-resources-node g/NodeID) ; for tests
   (input handler-infos g/Any :array)
+  (output handler-infos g/Any (g/fnk [handler-infos]
+                                (into [] (mapcat one-or-many-handler-infos-to-vec) handler-infos)))
```
Comment (matgis) [inline editor/src/clj/editor/gui.clj:2369]: I believe the `handler-infos` input is used by `defold/extension-spine` here, if you're wondering why it was added: https://github.com/defold/extension-spine/blob/main/defold-spine/editor/src/spineguiext.clj#L341

**Review by matgis — CHANGES_REQUESTED** [review]
Sadly we cannot use `g/node-value` from within an output without running the risk of caching stale values.

File: editor/src/clj/editor/gui.clj [hunk 3]
```diff
@@ -2566,6 +2566,12 @@
                                           (make-add-handler scene parent menu-label menu-icon add-fn opts)))))))]
       (filter some? (into node-options handler-options))))
 
+(defn- unused-display-profiles [gui-scene-node-id]
+  (g/with-auto-evaluation-context evaluation-context
+    (produce-unused-display-profiles
+      {:layout-msgs (g/node-value gui-scene-node-id :layout-msgs evaluation-context)
+       :display-profiles (g/node-value gui-scene-node-id :display-profiles evaluation-context)})))
+
```
Comment (matgis) [inline editor/src/clj/editor/gui.clj:2574]: It looks like this function could be replaced by just `(g/node-value scene :unused-display-profiles)` at the call site, or am I missing something?

Comment (vlaaad) [reply editor/src/clj/editor/gui.clj:2574]: Argh yes, you are totally right!

Comment (vlaaad) [reply editor/src/clj/editor/gui.clj]: Fixed!

**Review by matgis — APPROVED** [review]
Looks good to me!

**Pull request merged**